featuretools/entityset/entityset.py

```diff
@@ -280,231 +295,240 @@ def add_relationship(self,
 
         # _operations?
 
-        # this is a new pair of entities
-        child_e = relationship.child_dataframe
-        child_v = relationship.child_column.id
-        if child_e.index == child_v:
+        # this is a new pair of dataframes
+        child_df = relationship.child_dataframe
+        child_column = relationship._child_column_id
+        if child_df.ww.index == child_column:
             msg = "Unable to add relationship because child column '{}' in '{}' is also its index"
-            raise ValueError(msg.format(child_v, child_e.id))
-        parent_e = relationship.parent_dataframe
-        parent_v = relationship.parent_column.id
-        if not isinstance(child_e[child_v], vtypes.Id):
-            child_e.convert_variable_type(variable_id=child_v,
-                                          new_type=vtypes.Id,
-                                          convert_data=False)
-
-        if not isinstance(parent_e[parent_v], vtypes.Index):
-            parent_e.convert_variable_type(variable_id=parent_v,
-                                           new_type=vtypes.Index,
-                                           convert_data=False)
+            raise ValueError(msg.format(child_column, child_df.ww.name))
+        parent_df = relationship.parent_dataframe
+        parent_column = relationship._parent_column_id
+        if 'foreign_key' not in child_df.ww.semantic_tags[child_column]:
+            child_df.ww.add_semantic_tags({child_column: 'foreign_key'})
+
+        if parent_df.ww.index != parent_column:
+            parent_df.ww.set_index(parent_column)
         # Empty dataframes (as a result of accessing Entity.metadata)
         # default to object dtypes for discrete variables, but
         # indexes/ids default to ints. In this case, we convert
         # the empty column's type to int
-        if isinstance(child_e.df, pd.DataFrame) and \
-                (child_e.df.empty and child_e.df[child_v].dtype == object and
-                 is_numeric_dtype(parent_e.df[parent_v])):
-            child_e.df[child_v] = pd.Series(name=child_v, dtype=np.int64)
-
-        parent_dtype = parent_e.df[parent_v].dtype
-        child_dtype = child_e.df[child_v].dtype
-        msg = u"Unable to add relationship because {} in {} is Pandas dtype {}"\
-            u" and {} in {} is Pandas dtype {}."
-        if not is_dtype_equal(parent_dtype, child_dtype):
-            raise ValueError(msg.format(parent_v, parent_e.id, parent_dtype,
-                                        child_v, child_e.id, child_dtype))
+        # --> Implementation: is this still relevant?
+        if isinstance(child_df, pd.DataFrame) and \
+                (child_df.empty and child_df[child_column].dtype == object and
+                 parent_df.ww.columns[parent_column].is_numeric):
+            child_df.ww[child_column] = pd.Series(name=child_column, dtype=np.int64)
```

Comment:
The comment mentions metadata so this will come up when we tackle serialization.  The lines are covered on codecov main so this scenario does get tested pre-ww.  Looks like it has to do with ensuring that if the child dataframe was empty for some reason (the metadata uses empty dataframes to keep memory footprint small), the foreign key column on the child has an integer pandas dtype instead of an object dtype, to ensure the dtypes match between empty parent df and empty child df to pass the dtype validation check just below

Reply:
If I'm understanding this correctly - for Woodwork, as long as the parent column and the child column in the relationship have the same `LogicalType` we won't need to worry about manually doing this dtype conversion, as WW should handle it. If we create a new empty dataframe somewhere in the code, we will just need to make sure we set the correct logical types for the columns.

Reply:
Would the equivalent of an Entity with an empty dataframe just be a TableSchema, or are we we definitely going to need to have a Woodwork DataFrame where all columns have no data but they all have the correct dtypes for the schema?